### ISSUE TITLE
omitempty should remain for Cmd and Entrypoint

### ISSUE THREAD

**Issue by xh4n3** [fsouza/go-dockerclient#548]
Hi @fsouza @ybogdanov,

Sorry to pull this old issue out.
[omitempty for Cmd and Entrypoint](https://github.com/fsouza/go-dockerclient/issues/314)

Personally I think the solution is wrong. If anyone wants to rewrite a base image setting, that's a issue of base image, we probably should leave that option blank.
In addition, if removing omitempty is correct, we should also remove omitempty for other fields in the Config struct, e.g. Env, DNS.

So I suggest you @ybogdanov could considering to change your base image's setting instead of leaving these two fields Cmd and Entrypoint to be treated specially.

Thanks!

**Comment by ybogdanov** [comment]
@xh4n3 What about base images cannot change?

**Comment by xh4n3** [comment]
@ybogdanov How about you create your own image based on that, and override the entrypoint.

**Comment by ybogdanov** [comment]
@xh4n3 How you think Docker does itself so that you can override entrypoint? My case is the following: I'm developing [rocker](github.com/grammarly/rocker) which have to be compatible with `docker build` – that's it, it should be able to unset CMD and ENTRYPOINT. So this issue can only be solved in my case only if the API client will be able to do that.

**Comment by ybogdanov** [comment]
@xh4n3 Are you satisfied with the reasoning, can we close the issue? Or the selective "omitempty" solution is getting in your way somehow?

**Comment by xh4n3** [comment]
Sorry to be this late.
I've read your Rocker, and I know this could be impossible to work if omitempty remains.

The case here is, I wrote a kind of container manager tool just like Kubernetes.
I use this tool to push a job definition file to etcd, so we can start job from definitions in the etcd.
Definition file was written in YAMl, so tool would firstly unmarshall it into a struct to get stored in etcd. When starting a job, this tool would marshall the previously unmarshalled definition and pass it into go-dockerclient to start container.
So after the unmarshall and marshall processes, extra empty Cmd and Entrypoint were added, which then overrides the settings from Image.

So my temporary solution is stupid, checking in the source code of go-dockerclient and modified these two fields.
